cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetInput/index.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import withStyles, { StyleRules, WithStyles } from '@material-ui/core/styles/withStyles';
+import If from 'components/If';
+import { WIDGET_TYPES } from 'components/PluginJSONCreator/constants';
+import PluginInput from 'components/PluginJSONCreator/Create/Content/PluginInput';
+import { ICreateContext } from 'components/PluginJSONCreator/CreateContextConnect';
+import * as React from 'react';
+
+const styles = (): StyleRules => {
+  return {
+    widgetInput: {
+      '& > *': {
+        width: '100%',
+        marginTop: '10px',
+        marginBottom: '10px',
+      },
+    },
+  };
+};
+
+interface IWidgetInputProps extends WithStyles<typeof styles>, ICreateContext {
+  widgetID: number;
+}
+
+const WidgetInputView: React.FC<IWidgetInputProps> = ({
+  classes,
+  widgetID,
+  widgetToInfo,
+  setWidgetToInfo,
+}) => {
+  function onNameChange(obj, id) {
+    return (name) => {
+      setWidgetToInfo((prevObjs) => ({ ...prevObjs, [id]: { ...obj, name } }));
+    };
+  }
+
+  function onLabelChange(obj, id) {
+    return (label) => {
+      setWidgetToInfo((prevObjs) => ({ ...prevObjs, [id]: { ...obj, label } }));
+    };
+  }
+
+  function onWidgetTypeChange(obj, id) {
+    return (widgetType) => {
+      setWidgetToInfo((prevObjs) => ({ ...prevObjs, [id]: { ...obj, widgetType } }));
+    };
+  }
+
+  function onWidgetCategoryChange(obj, id) {
+    return (widgetCategory) => {
+      setWidgetToInfo((prevObjs) => ({ ...prevObjs, [id]: { ...obj, widgetCategory } }));
+    };
+  }
+
+  const widgetInfo = widgetToInfo[widgetID];
+
+  return (
+    <If condition={widgetInfo}>
+      <div className={classes.widgetInput}>
+        <PluginInput
+          widgetType={'textbox'}
+          value={widgetInfo.name}
+          setValue={onNameChange(widgetInfo, widgetID)}
+          label={'Name'}
+          placeholder={'Name a Widget'}
+          required={false}
+        />
+        <PluginInput
+          widgetType={'textbox'}
+          value={widgetInfo.label}
+          setValue={onLabelChange(widgetInfo, widgetID)}
+          label={'Label'}
+          placeholder={'Label a Widget'}
+          required={false}
+        />
+        <PluginInput
+          widgetType={'textbox'}
```

Comment:
nit: Is the category `plugin`? Can this be a dropdown with only one value. We really don't want the user to type this in.

Reply:
changed!

Now it is a `select` widget with `options=['plugin']`.